src/zenml/models/v2/base/scoped.py

```diff
@@ -170,6 +187,73 @@ def set_scope_user(self, user_id: UUID) -> None:
         """
         self.scope_user = user_id
 
+    def get_custom_filters(
+        self,
+    ) -> List["ColumnElement[bool]"]:
+        """Get custom filters.
+
+        Returns:
+            A list of custom filters.
+        """
+        custom_filters = super().get_custom_filters()
+
+        from sqlmodel import and_
+
+        from zenml.zen_stores.schemas import (
+            PipelineSchema,
+            UserSchema,
+        )
+
+        if self.user:
+            user_filter = and_(
+                PipelineSchema.user_id == UserSchema.id,
```

Comment:
Good catch, this is a complete oversight from my side. I will fix this right away.

Reply:
I have changed the base implementation as follows per your suggestion:

```python
    def get_custom_filters(
        self, table: Type["AnySchema"]
    ) -> List["ColumnElement[bool]"]:
        """Get custom filters.

        This can be overridden by subclasses to define custom filters that are
        not based on the columns of the underlying table.

        Args:
            table: The query table.

        Returns:
            A list of custom filters.
        """
```